connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -217,6 +218,28 @@ message Expression {
     bool is_user_defined_function = 4;
   }
 
+  message ScalarInlineUserDefinedFunction {
```

Comment:
I had a discussion with @hvanhovell and we agreed that the message type should be moved out of the expression message because we want to use the same message at the end of the day for registered and inline function.

Reply:
Sounds good. How about the new ScalarInlineUserDefinedFunction as shown below? We will serialize the existing PythonUDF message to `language_sepecific` field then. Please let me know if that's what we want. @hvanhovell @grundprinzip 

```
  message ScalarInlineUserDefinedFunction {

    string function_name = 1;

    bool deterministic = 2;

    repeated Expression arguments = 3;

    FunctionLanguage language = 4;

    bytes language_sepecific = 5
  }
```

Reply:
Actually no, the message that you have is almost perfect. Please simply move it out of the expression message and that's all for now. Herman had some other ideas. I will comment back on Monday.

Reply:
I see. Then should `ScalarInlineUserDefinedFunction` belong to the Command message type? I am not clear how SparkConnectPlanner should process `ScalarInlineUserDefinedFunction` if it doesn't belong to any higher-level message type: Relation, Expression, or Command.

Reply:
Right now the message is nested inside expression.

```
message Expression {
   oneof type {
   ScalarInlineUserDefinedFunction function = 1;
   }

   message ScalarInlineUserDefinedFunction {
   }
}
```

Just move it outside, the nesting is really just for convenience and intend and that is why it makes sense to move it outside of expression because it can be used in both variants.

```
message Expression {
   oneof type {
   ScalarInlineUserDefinedFunction function = 1;
   } 
}

message ScalarInlineUserDefinedFunction {
}
```

Reply:
Makes sense! Refactored.